### ISSUE TITLE
allow for use of metadata.json in place of metadata.rb

### ISSUE THREAD

**Issue by jtimberman** [Foodcritic/foodcritic#352]
When uploading a cookbook to [Supermarket](https://supermarket.chef.io), the site's app runs foodcritic against the cookbook. This is cool, it shows up on a "foodcritic" tab for the cookbook's page so users can see if it passes lint checks or not.

However, when uploading cookbooks with [stove](https://github.com/sethvargo/stove), the metadata.rb is stripped from the uploaded tarball. This causes the foodcritic check to fail with `FC031` and `FC045`. For example, the  [plexapp cookbook](https://supermarket.chef.io/cookbooks/plexapp/versions/1.0.0#foodcritic):

```
FC031: Cookbook without metadata file: /tmp/cook/1a37f43b65668fe3c93da7f4/plexapp/metadata.rb:1
FC045: Consider setting cookbook name in metadata: /tmp/cook/1a37f43b65668fe3c93da7f4/plexapp/metadata.rb:1
```

Seth has a [detailed explanation about his rationale](https://github.com/sethvargo/stove/pull/59#issuecomment-61488239). I don't have an argument about that. Certainly a user can add a `.foodcritic` config file to ignore those rules, but that means every single cookbook needs that. I think it would be better if Foodcritic would first check the `metadata.json` exists, then check that the name is in that metadata file. 

I also think it would be nice if there were a separate rule that validates that if both `metadata.json` and `metadata.rb` files exist in a cookbook, that rendering the `metadata.json` again from `metadata.rb` results in the same output. I can open a separate issue for that, though.

**Comment by tas50** [comment]
I'm working on this one specifically for Supermarket

**Comment by tas50** [comment]
I'm closing this one out. I looked into this and it requires a fundamental rearchitecture of how Foodcritic works. We do AST parsing and we've built an enormous number of helpers around parsing the AST of the metadata. If we switch to JSON the format of the file is actually different. We'd have to build a ton of logic into those helpers to allow them to function with both. I went down that road and then pretty quickly realized it was sadness everywhere.
